homeassistant/components/mobile_app/__init__.py

```diff
@@ -96,3 +102,34 @@
         )
 
     return True
+
+
+async def async_unload_entry(hass, entry):
+    """Unload a mobile app entry."""
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(entry, component)
+                for component in PLATFORMS
+            ]
+        )
+    )
+    if not unload_ok:
+        return False
+
+    webhook_unregister(hass, entry.data[CONF_WEBHOOK_ID])
+
+    return True
+
+
+async def async_remove_entry(hass, entry):
+    """Cleanup when entry is removed."""
+    hass.data[DOMAIN][DATA_DELETED_IDS].append(entry.data[CONF_WEBHOOK_ID])
```

Comment:
When is this list cleared?

Reply:
It never is. We keep track for a better user experience.

Webhooks will always return a 200 so you don't know if the data was received. To differentiate from a "webhook works" vs "registration deleted", we send a 404 if a deleted ID is hit.

Reply:
I guess we're not worried about a memory leak since config entries are manually added and removed?

Reply:
Yeah. It will take a while before we fill this with too many. I'll make a note to add timestamps to this and purge old ones in a future PR (this is existing code).